### ISSUE TITLE
[Typescript] Prettier removes question mark from optional computed class property

### ISSUE THREAD

**Issue by spencersutton** [prettier/prettier#6569]
**Prettier 1.18.2**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuc0DOMAEAjAhgJ0wF5MByPfUgbgB0owAbXNNTAMQgk2Dsz8wDaFALoB+WlH6CAjGIlSB5XAC9ScugF86IADQgIABxgBLdMlAF8EAO4AFAgjTIQuAG4RjAE10hs+XGAA1nAwAMoGAcZQAObIMPgArnB6ABYwALYMAOopxvBoEWBwoY55xq55AJ7OYCw+UWhw+DC2-tHpuMgAZrgMjXoAVmgAHgBC-kEhobjpcAAyUXDdvf0gQ8OhUdEMcACKCRDwy33JIBH4jfjOMJUGcGhg+MZGPgZPsFleMCnIABwADHo3hBGll-AZnG97k1XEs9ABHA7wVqGJwoZgAWigcDgnlxPnwcERxkJrVw7U6SB6Jz0jXSxjiiVOaC2O32hyWVJWpxguGwn0832QACY9PFcMYGFsAMIQdIdZxQaBwkAJRoAFT5aOpjQ0GiAA)
```sh
--parser typescript
```

**Input:**
```tsx
const bar = 'bar';
class Foo {
    [bar]?;
    [1]?;
    ['baz']?;
}

```

**Output:**
```tsx
const bar = "bar";
class Foo {
  [bar];
  [1]?;
  ["baz"]?;
}

```

**Expected behavior:**
Prettier should preserve ? in output. Expected output:
```tsx
const bar = "bar";

class Foo {
  [bar]?;
  [1]?;
  ["baz"]?;
}

```

This only happens when the computed property is a variable.

**Comment by fisker** [comment]
minimum reproduct

**Prettier 1.18.2**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuc0DOMAEAjAhgJ0wF5MByPfUgbgB0owAbXNNTAMQgk2Dsz8wDaFALoB+WlAC+dEABoQEAA4wAlumSgC+CAHcACgQRpkIXADcIKgCZyQ2fLjABrODADKixyqgBzZDHwAVzh5AAsYAFsGAHVQlXg0TzA4NyN4lTN4gE8TMBZbbzQ4fBg9Bx8I3GQAM1wGIvkAKzQADwAhB2dXN1wIuAAZbzgauoaQZpa3bx8GOABFQIh4EfqQkE98IvwTGCzFODQwfBVlW0Vj2GjrGFDkAA4ABnlziCLoh0UTc4Pis2H5ACOi3gZSUxhQzAAtFA4HArHDbPg4ECVEiyrgKlUkLVVvIihEVP4gms0NNZgslsNsaM1jBcNgrlYbsgAEzyAK4FQMaYAYQgEUqJig0H+IECRQAKvTwTiipJJEA)
```sh
--parser typescript
```

**Input:**
```tsx
const bar = 'bar';
class Foo {
    [bar]?;
}

```

**Output:**
```tsx
const bar = "bar";
class Foo {
  [bar];
}

```

**Expected behavior:**
```tsx
const bar = "bar";
class Foo {
  [bar]?;
}

```

**Comment by Lodin** [comment]
@fisker, for me, it breaks the code completely moving the question mark inside the brackets:
[Playground repro](https://prettier.io/playground/#N4Igxg9gdgLgprEAuc0DOMAEbMF5MDKAngLYBGEANgBQCUA3ADpTNiUCGaOAgpsM5kwAHAE4R4YeABNMAbTQBdAPx0+AwZnUBfZlpAAaEBCEwAlumSh2IsQHcACtYRpkIdgDcIpqQZBkR7GAA1nAwBEKBplAA5sgwIgCucIYAFjAklADqKabwaBFgcATOuabuuUSuYFy+UWhwIjD2AdEk7MgAZuyU9YYAVmgAHgBCAcGhBOwkcAAyUXCd3b0gA4MEUdGUcACKCeILSF09ySARIvUirjBEQnBoYCKmJr6iUTCZ3jApyAAcAAyGUQQeqZAJCVyiO4NdwLQwARz28GaxhcKE4AFooHA4FIcb4RHAEaYCc12K12oclid6iRTHFEtSNltdvt6UlDDB2GQPlIvsgAEwcgKmSgbADCEBIbVcUGgsJACXqABUuaj4uy3BhFsdDFIIGBtcsIAkYEITfzDXAtFogA)

Input:
```typescript
const s = Symbol();

class A {
  protected [s]?() {
    
  }
}
```

Output:
```typescript
const s = Symbol();

class A {
  protected [s?]() {}
}
```

**Comment by sosukesuzuki** [comment]
@Lodin Probably, your issue will be fixed with #6673:+1:

**Comment by evilebottnawi** [comment]
Merged https://github.com/prettier/prettier/pull/6673, i think we should add more tests cases from issue and close it

**Comment by sosukesuzuki** [comment]
@evilebottnawi The original problem is different from the issue fixed with #6673. I think we cannot close this issue without updating the `typescript-estree`.

**Comment by thorn0** [comment]
As I wrote in #6601:

> #6673 fixes @Lodin 's issue, but only for simple cases where the key is a simple identifier (variable). Cases where the key is a more complex expression indeed require an update of typescript-estree to be fixed.
